### PR TITLE
[30] Topics index action

### PR DESCRIPTION
**Pull request by juanmanuelramallo**

### CODE REVIEW

File: app/controllers/topics_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class TopicsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    render json: Topic.all
```
Comment (Alxzu) [inline app/controllers/topics_controller.rb]: This is fine, maybe for next prs you can add pagination, maybe this one? https://github.com/ddnexus/pagy

Comment (juanmanuelramallo) [reply app/controllers/topics_controller.rb]: Sure, maybe not for topics but for conversations or targets?
I believe topics are just 9, and the user cannot add/delete them

Comment (Alxzu) [reply app/controllers/topics_controller.rb]: Yeah, if that's the case don't bother to add pagination on topics

Comment (juanmanuelramallo) [reply app/controllers/topics_controller.rb]: https://trello.com/c/BZ7aXHLo/31-add-pagination-to-targets-index-action
https://trello.com/c/UkuF4YS0/32-add-pagination-to-conversations-index-action
Created these two.

File: spec/requests/topics/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'GET /topics', type: :request do
+  let(:user) { create :user }
+
+  subject do
+    get '/topics', headers: headers
```
Comment (GuilleLeopold) [inline spec/requests/topics/index_spec.rb]: remember you can use the helper instead of '/topics'

Comment (juanmanuelramallo) [reply spec/requests/topics/index_spec.rb]: I completely forgot, I'll update.

File: spec/requests/topics/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    data
+  end
+
+  before { create_list :topic, 3 }
+
+  it 'returns 3 topics' do
```
Comment (GuilleLeopold) [inline spec/requests/topics/index_spec.rb]: You can test that the topics returned are the one that you have created by verifying its information. So you make sure you are not returning three times the same topic or whatever, in this case it's really simple, you are just doing Topic.all but take it into account for the future

**Pull request merged**